packages/mds-agency/api.ts

```diff
@@ -727,16 +727,20 @@ function api(app: express.Express): express.Express {
     return null
   }
 
-  async function writeTelemetry(telemetry: Telemetry | Telemetry[]): Promise<void[]> {
+  async function writeTelemetry(telemetry: Telemetry | Telemetry[]) {
     if (!Array.isArray(telemetry)) {
-      const promises = [
+      const promises: (Promise<any>)[] = [
```

Comment:
I'm gonna clone this down and see if I can find a better way to express the type than this hack we came up with earlier today... Seems super odd that we'd need to use the `Promise<any>[]` here. WE WILL REBUILD IT! MAKE IT STRONGER! FASTER!

Reply:
Try to use destructuring and let TS infer the types `const [x, y, z] = [...]`or declare the type as a tuple `const promises: [Promise<x>, Promise<y>, Promise<z>] = ...`

Reply:
```
async function writeTelemetry(telemetry: Telemetry | Telemetry[]) {
    if (!Array.isArray(telemetry)) {
      const promises: [Promise<number>, Promise<void>, Promise<void>] = [
        db.writeTelemetry([telemetry]),
        cache.writeTelemetry([telemetry]),
        stream.writeTelemetry([telemetry])
      ]
      return Promise.all(promises)
    }
    const promises: [Promise<number>, Promise<void>, Promise<void>] = [
      db.writeTelemetry(telemetry),
      cache.writeTelemetry(telemetry),
      stream.writeTelemetry(telemetry)
    ]
    return Promise.all(promises)
  }
```

seems to work!

Reply:
check it in!